### DISCUSSION TITLE
How to get the mass fraction in a phase

### DISCUSSION THREAD

**Discussion by Noxical** [reaktoro/reaktoro#288] [General]
I want to retrieve the mass fractions in a phase.

However in reaktoro its seem only speciesMoleFractions is available. 

To find mass fraction therefore I use for example: 

(props.speciesMass('H2)')/ (props.speciesMass('H2')+props.speciesMass('Fe(cd)')

However for many components this because comberstone. So I wanted to use props.phaseMass() but this doens't seem to work.

How to calculate the mass fractions most conveniently?

**Comment by allanleal** [comment]
Hi @Noxical , a `speciesMassFraction` method in `ChemicalProps` could indeed be an interesting addition.

For now, to get the mass of a phase, you can use `props.phaseProps("PhaseName").mass()`. To get the mass of the system (i.e., the sum of the masses of all species; equivalent to the sum of the masses of all phases), use `props.mass()`.

**Reply by Noxical** [reply]
thanks @allanleal , this will work for now. 
Note that in many metallurgical calculations the activities are often expressed in mass fraction / percentage compared to atom fractions. So to make it work with reaktoro I am changing the activities from mass to moles. But to check I need to convert back.

**Reply by allanleal** [reply]
Interesting. Thanks for sharing this. So, have you been defining new activity models in Reaktoro? I have not tried doing this from Python yet. Are you using C++?

**Reply by Noxical** [reply]
Yes I am working on the activity models, I am writing them in python for now. When finished I will convert to C++ for implementation in reaktoro. At that point I will probably want to have discussion with you first if possible.

**Reply by allanleal** [reply]
Definitely. I can point you in the right directions.
